src/transforms/transform-pull-request.ts

```diff
@@ -18,30 +18,55 @@ const mapStatus = (status: string, merged_at?: string) => {
 	return "UNKNOWN";
 };
 
+interface JiraReviewer extends JiraReview {
+	login?: string;
+}
+
 // TODO: define arguments and return
-const mapReviews = (reviews: Octokit.PullsListReviewsResponse = []) => {
+const mapReviews = async (reviews: Octokit.PullsListReviewsResponse = [], gitHubInstallationClient: GitHubInstallationClient): Promise<JiraReview[]> => {
 	const sortedReviews = orderBy(reviews, "submitted_at", "desc");
-	const usernames: Record<string, JiraReview> = {};
-	// The reduce function goes through all the reviews and creates an array of unique users (so users' avatars won't be duplicated on the dev panel in Jira) and it considers 'APPROVED' as the main approval status for that user.
-	return sortedReviews.reduce((acc: JiraReview[], review) => {
+	const usernames: Record<string, JiraReviewer> = {};
+	// The reduce function goes through all the reviews and creates an array of unique users
+	// (so users' avatars won't be duplicated on the dev panel in Jira)
+	// and it considers 'APPROVED' as the main approval status for that user.
+	const reviewsReduced: JiraReviewer[] = sortedReviews.reduce((acc: JiraReviewer[], review) => {
 		// Adds user to the usernames object if user is not yet added, then it adds that unique user to the accumulator.
 		const author = review?.user;
+
 		if (!usernames[author?.login]) {
 			usernames[author?.login] = {
 				...getJiraAuthor(author),
+				login: author.login,
 				approvalStatus: review?.state === "APPROVED" ? "APPROVED" : "UNAPPROVED"
 			};
+
 			acc.push(usernames[author?.login]);
-			// If user is already added (not unique) but the previous approval status is different than APPROVED and current approval status is APPROVED, updates approval status.
+			// If user is already added (not unique) but the previous approval status is different
+			// from APPROVED and current approval status is APPROVED, updates approval status.
 		} else if (
 			usernames[author?.login].approvalStatus !== "APPROVED" &&
 			review.state === "APPROVED"
 		) {
 			usernames[author?.login].approvalStatus = "APPROVED";
 		}
+
 		// Returns the reviews' array with unique users
 		return acc;
 	}, []);
+
+	// Get GitHub user email, so it can be matched to an AAID
+	return Promise.all(reviewsReduced.map(async reviewer => {
+		let login;
+		if (reviewer.login) {
+			login = reviewer.login;
+			delete reviewer.login;
```

Comment:
Please let me know if there's a better way to do this, I've explored a number of options but would like your input :)

Reply:
well, you could just do 
```
const gitHubUser = await getGithubUser(gitHubInstallationClient, login);
delete reviewer.login;
return {
			...reviewer,
			email: gitHubUser?.email || `${login}@noreply.user.github.com`
		};
```

Reply:
Not really because `reviewer.login` is optional and the signature of `getGithubUser` expects login. I don't think we want to change that to not be required?